### PR TITLE
Dice app for java

### PR DESCRIPTION
**Pull request by svrnm**
See https://github.com/open-telemetry/opentelemetry.io/issues/2623

Preview: https://deploy-preview-2625--opentelemetry.netlify.app/docs/instrumentation/java/getting-started/

### CODE REVIEW

File: content/en/docs/instrumentation/java/getting-started.md [hunk 1]
```diff
@@ -1,138 +1,216 @@
...
 
-## Run the instrumented example
+## Example Application
+
+The following example uses a basic spring boot application.
+
+### Dependencies
+
+Create a folder called `java-simple` and within that folder, create a file
+called `build.gradle` with the following content:
```
Comment (mateuszrzeszutek) [inline content/en/docs/instrumentation/java/getting-started.md:28]:
```suggestion
called `build.gradle.kts` with the following content:
```

File: content/en/docs/instrumentation/java/getting-started.md [hunk 2]
```diff
@@ -1,138 +1,216 @@
...
+
+repositories {
+  mavenCentral()
+}
+
+dependencies {
+  implementation 'org.springframework.boot:spring-boot-starter-web',
+                 'org.apache.logging.log4j:log4j-core:2.17.2'
+}
+```
```
Comment (mateuszrzeszutek) [inline content/en/docs/instrumentation/java/getting-started.md:50]:
```suggestion
```kotlin
plugins {
  id("java")
  id("org.springframework.boot") version "3.0.6"
  id("io.spring.dependency-management") version "1.1.0"
}

sourceSets {
  main {
    java.setSrcDirs(setOf("."))
  }
}

repositories {
  mavenCentral()
}

dependencies {
  implementation("org.springframework.boot:spring-boot-starter-web")
}
```
```

File: content/en/docs/instrumentation/java/getting-started.md [hunk 3]
```diff
@@ -1,138 +1,216 @@
...
+import org.springframework.boot.Banner;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+@SpringBootApplication
+public class DiceApplication {
+  public static void main(String[] args) {
+    SpringApplication app = new SpringApplication(DiceApplication.class);
+    app.setBannerMode(Banner.Mode.OFF);
+    app.run(args);
+  }
+}
```
Comment (mateuszrzeszutek) [inline content/en/docs/instrumentation/java/getting-started.md:69]:
```suggestion
package otel;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.Banner;
import org.springframework.boot.autoconfigure.SpringBootApplication;

@SpringBootApplication
public class DiceApplication {
  public static void main(String[] args) {
    SpringApplication app = new SpringApplication(DiceApplication.class);
    app.setBannerMode(Banner.Mode.OFF);
    app.run(args);
  }
}
```

File: content/en/docs/instrumentation/java/getting-started.md [hunk 4]
```diff
@@ -1,138 +1,216 @@
...
+      logger.info(player.get() + " is rolling the dice: " + result);
+    } else {
+      logger.info("Anonymous player is rolling the dice: " + result);
+    }
+    return Integer.toString(result);
+  }
+    public int getRandomNumber(int min, int max) {
+        return (int) ((Math.random() * (max - min)) + min);
+    }
+}
```
Comment (mateuszrzeszutek) [inline content/en/docs/instrumentation/java/getting-started.md:101]:
```suggestion
package otel;

import java.util.Optional;
import java.util.concurrent.ThreadLocalRandom;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class RollController {
  private static final Logger logger = LoggerFactory.getLogger(RollController.class);

  @GetMapping("/rolldice")
  public String index(@RequestParam("player") Optional<String> player) {
    int result = this.getRandomNumber(1, 6);
    if (player.isPresent()) {
      logger.info("{} is rolling the dice: {}", player.get(), result);
    } else {
      logger.info("Anonymous player is rolling the dice: {}", result);
    }
    return Integer.toString(result);
  }

  public int getRandomNumber(int min, int max) {
    return ThreadLocalRandom.current().nextInt(min, max + 1);
  }
}
```

File: content/en/docs/instrumentation/java/getting-started.md [hunk 5]
```diff
@@ -1,138 +1,216 @@
...
+```
+
+To test the application, build and run it:
+
+```console
+gradle assemble
+java -jar ./java-simple.jar
```
Comment (mateuszrzeszutek) [inline content/en/docs/instrumentation/java/getting-started.md:108]:
```suggestion
java -jar ./build/libs/java-simple.jar
```

File: content/en/docs/instrumentation/java/getting-started.md [hunk 6]
```diff
@@ -1,138 +1,216 @@
...
+     OTEL_LOGS_EXPORTER=logging
+   ```
+
+   {{% alert title="Important" color="warning" %}}Replace `PATH/TO` above, with
+   your path to the JAR.{{% /alert %}}
+
+3. Run your **application** once again:
+
+   ```console
+   $ java -jar ./java-simple.jar
```
Comment (mateuszrzeszutek) [inline content/en/docs/instrumentation/java/getting-started.md:145]:
```suggestion
   $ java -jar ./build/libs/java-simple.jar
```

File: content/en/docs/instrumentation/java/getting-started.md [hunk 7]
```diff
@@ -1,138 +1,216 @@
...
+
+   {{% alert color="info" %}}<i class="fas fa-edit"></i> Take note of the path
+   to the JAR file.{{% /alert %}}
+
+2. Set and export variables that specify the Java agent JAR and a [console
+   exporter][], using a notation suitable for your shell/terminal environment
+   &mdash; we illustrate a notation for bash-like shells:
+
+   ```console
+   $ export JAVA_OPTS="-javaagent:PATH/TO/opentelemetry-javaagent.jar" \
```
Comment (mateuszrzeszutek) [inline content/en/docs/instrumentation/java/getting-started.md:133]: AFAIK `JAVA_TOOL_OPTIONS` is the universally recognized env var name -- at least that what's been working for me (temurin 17)
```suggestion
   $ export JAVA_TOOL_OPTIONS="-javaagent:PATH/TO/opentelemetry-javaagent.jar" \
```

File: content/en/docs/instrumentation/java/getting-started.md [hunk 8]
```diff
@@ -146,19 +224,24 @@ For more:
 - Learn about [manual instrumentation][] and try out more
   [examples]({{% relref examples %}}).
 
+## Troubleshooting
```
Comment (mateuszrzeszutek) [inline content/en/docs/instrumentation/java/getting-started.md:227]: There already is a troubleshooting section here: https://opentelemetry.io/docs/instrumentation/java/automatic/#troubleshooting

**Review by mateuszrzeszutek — COMMENTED** [review]
I ran your example, and decided to fix a couple of things:

* Used Kotlin instead of Groovy for Gradle scripts (Groovy is not exactly a great tool, especially for gradle scripts, and I feel bad about promoting it in our docs)
* Removed log4j dependency, spring-starter-web already contains slf4j & logback
* Simplified the dice calculation
* Reformatted everything
* Fixed the ourput jar path (gradle places built artifacts in `build/libs/`)

Overall, I think it's a great idea to have a self-contained example in our docs 👍

**Comment by svrnm** [comment]
> I ran your example, and decided to fix a couple of things:
> 
> * Used Kotlin instead of Groovy for Gradle scripts (Groovy is not exactly a great tool, especially for gradle scripts, and I feel bad about promoting it in our docs)
> * Removed log4j dependency, spring-starter-web already contains slf4j & logback
> * Simplified the dice calculation
> * Reformatted everything
> * Fixed the ourput jar path (gradle places built artifacts in `build/libs/`)
> 
> Overall, I think it's a great idea to have a self-contained example in our docs 👍

wow, thank you!

File: content/en/docs/instrumentation/java/getting-started.md (hunk 7 above)
Comment (svrnm) [reply content/en/docs/instrumentation/java/getting-started.md:138]: Good call, I ran into that issue myself but forgot to fix that!

File: content/en/docs/instrumentation/java/getting-started.md (hunk 8 above)
Comment (svrnm) [reply content/en/docs/instrumentation/java/getting-started.md:231]: I put it here to be consistent with the JavaScript documentation, but let me remove it for now, we don't need to have it twice

**Review by mateuszrzeszutek — APPROVED** [review]
Thanks!

**Review by cartermp — APPROVED** [review]

**Pull request merged**